python/pydantic_core/_pydantic_core.pyi

```diff
@@ -606,6 +606,18 @@ class Url(SupportsAllComparisons):
             An instance of URL
         """
 
+    def join(self, path: str, trailing_slash: bool = True) -> Self:
```

Comment:
Is `True` the right default here? Part of me thinks that if joining files like `a.png` (which might be quite common) that this default is unhelpful, especially as user can also get a trailing slash by passing it to the input.

Also, I think I would prefer the name `append_trailing_slash` as this doesn't strip them:

```suggestion
    def join(self, path: str, append_trailing_slash: bool = False) -> Self:
```